### PR TITLE
ThreadChannel#setLocked fix

### PR DESCRIPTION
**Pull request by ErdemGKSL**
In order to lock the channel, you need to archive it at the same time according to discord API.

### CODE REVIEW

**Comment by Jiralite** [comment]
I don't think this is correct.

You can lock a request, sure, and nothing happens visibly. However, closing it normally will actually lock the thread too because `locked` is `true`.

Look at what happens in the client. There's no "Lock" option, only "Close & Lock". Discord is setting both `archived` and `lock` to `true` here.

If you want to close and lock a thread, you should close and lock the thread - not only lock it.

**Comment by ErdemGKSL** [comment]
So that would make more sense right? I was confused because I was judging the API by the frontend requests actually sorry for my mistake. Wouldn't such use be more logical so that others do not confuse it? Or it can be in setLockedAndArchived in order to maintain systematicity.

**Comment by ImRodry** [comment]
If your thread is archived and you want to lock it, you need to unarchive and then archive and lock at the same time. If it is open and you want to lock it you have to archive and lock like is being done in this PR, I just don't know if this is all that useful, would just be another shortcut to something you could easily do yourself

**Comment by jaw0r3k** [comment]
and also it need typings

File: packages/discord.js/src/structures/ThreadChannel.js [hunk 1]
```diff
@@ -568,15 +584,15 @@ class ThreadChannel extends BaseChannel {
...
-  bulkDelete() {}
+  get lastMessage() { }
+  get lastPinAt() { }
+  send() { }
+  sendTyping() { }
+  createMessageCollector() { }
+  awaitMessages() { }
+  createMessageComponentCollector() { }
+  awaitMessageComponent() { }
+  bulkDelete() { }
```
Comment (NotSugden) [inline packages/discord.js/src/structures/ThreadChannel.js:595]:
```suggestion
  get lastMessage() {}
  get lastPinAt() {}
  send() {}
  sendTyping() {}
  createMessageCollector() {}
  awaitMessages() {}
  createMessageComponentCollector() {}
  awaitMessageComponent() {}
  bulkDelete() {}
```

File: packages/discord.js/src/structures/ThreadChannel.js [hunk 2]
```diff
@@ -422,6 +422,22 @@ class ThreadChannel extends BaseChannel {
...
+   * {@link PermissionFlagsBits.ManageThreads} permission can unarchive it.
+   * @param {string} reason Reason for locking or unlocking the thread
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Set the thread to locked and archived
+   * thread.closeAndLock()
+   *   .then(() => console.log(`Thread is now locked and archived.`))
+   *   .catch(console.error);
+   */
+  closeAndLock(reason) {
```
Comment (NotSugden) [inline packages/discord.js/src/structures/ThreadChannel.js:437]: not a fan of the name.

i think a better soloution would be to add a `lock` parameter to setArchive?

**Review by NotSugden — CHANGES_REQUESTED** [review]
typings required

File: packages/discord.js/src/structures/ThreadChannel.js (hunk 2 above)
Comment (ckohen) [reply packages/discord.js/src/structures/ThreadChannel.js:437]: I agree setArchived should have three potential signatures,
setArchived() => archives
setArchived(false, lock: never) => unarchived
setArchived(true, lock: boolean) => archive and set lock accordingly

**Pull request closed without merging**